areader/reader.go

```diff
@@ -160,6 +160,9 @@ func (ar *Reader) readHeader(headerSum []byte, comp artifact.Compressor) error {
 		return err
 	}
 
+	// Empty the remaining reader
+	io.Copy(ioutil.Discard, r)
```

Comment:
Nice fix!

The fact that it needs to be called here is quite surprising though, it would be nice to have a ticket reference here for later, to understand why it is needed. Same for below.

Reply:
So, I'm not really happy with this fix tbh. But the checksum reader and the verify function is sort of entangled too, so, yeh. this was the least intrusive fix I could find. But it does not fix this error for other parts using the checksum reader...

I don't know really..

Maybe I should dive deeper with this (but the reader code is messy :anger: )

Reply:
> What is the bytes left over in the reader?

The header.tar.gz is 514 bytes, with the last two bytes being 00 00. Sometimes the read was 514, other times the read was 512.

Reply:
> No test, I haven't been able to reproduce.

It is not reproducible in standalone, only via daemon downloading from AWS s3.

Reply:
@grandfield: And can you reproduce it with this fix?

@oleorhagen: The fact that it's an elusive bug is why I thought a comment would be good. It at least reduces the chance that we will remove it by mistake. I think it will be hard to test this when it's not consistently reproducible.

Reply:
Added the comment and the changelog.

Reply:
> @grandfield: And can you reproduce it with this fix?
Works for me consistently now.